src/gdk2x.nim

```diff
@@ -0,0 +1,249 @@
+      dynlib: lib, importc: "gdk_xid_table_lookup".}
+    proc net_wm_supports*(property: gdk2.PAtom): gboolean {.cdecl,
+      dynlib: lib, importc: "gdk_net_wm_supports".}
+  proc x11_grab_server*() {.cdecl,
+    dynlib: lib, importc: "gdk_x11_grab_server".}
+  proc x11_ungrab_server*() {.cdecl,
+    dynlib: lib, importc: "gdk_x11_ungrab_server".}
+
+proc x11_lookup_xdisplay*(xdisplay: xlib.PDisplay): gdk2.PDisplay {.cdecl,
+    dynlib: lib, importc: "gdk_x11_lookup_xdisplay".}
+proc x11_atom_to_xatom_for_display*(display: gdk2.PDisplay,
+    atom: gdk2.PAtom): x.PAtom {.cdecl,
+      dynlib: lib, importc: "gdk_x11_atom_to_xatom_for_display".}
+proc x11_xatom_to_atom_for_display*(display: gdk2.PDisplay,
+    xatom: x.PAtom): gdk2.PAtom {.cdecl,
+      dynlib: lib, importc: "gdk_x11_xatom_to_atom_for_display".}
+proc x11_get_xatom_by_name_for_display*(display: gdk2.PDisplay,
+    atom_name: PPgchar): x.PAtom {.cdecl,
+      dynlib: lib, importc: "gdk_x11_get_xatom_by_name_for_display".}
+proc x11_get_xatom_name_for_display*(display: gdk2.PDisplay,
+    xatom: x.PAtom): PPgchar {.cdecl,
+      dynlib: lib, importc: "gdk_x11_get_xatom_name_for_display".}
+
+when not defined(MULTIHEAD_SAFE):
+  proc x11_atom_to_xatom*(atom: gdk2.PAtom): x.PAtom {.cdecl,
+    dynlib: lib, importc: "gdk_x11_atom_to_xatom".}
+  proc x11_xatom_to_atom*(xatom: x.PAtom): gdk2.PAtom {.cdecl,
+    dynlib: lib, importc: "gdk_x11_xatom_to_atom".}
+  proc x11_get_xatom_by_name*(atom_name: PPgchar): x.PAtom {.cdecl,
+    dynlib: lib, importc: "gdk_x11_get_xatom_by_name".}
+  proc x11_get_xatom_name*(xatom: x.PAtom): PPgchar {.cdecl,
+    dynlib: lib, importc: "gdk_x11_get_xatom_name".}
+
+
+proc x11_display_grab*(display: gdk2.PDisplay) {.cdecl,
+    dynlib: lib, importc: "gdk_x11_display_grab".}
+proc x11_display_ungrab*(display: gdk2.PDisplay) {.cdecl,
+    dynlib: lib, importc: "gdk_x11_display_ungrab".}
+proc x11_register_standard_event_type*(display: gdk2.PDisplay,
+    event_base: gint, n_events: gint) {.cdecl,
+      dynlib: lib, importc: "gdk_x11_register_standard_event_type".}
+
+when not defined(DISABLE_DEPRECATED) or defined(COMPILATION):
+  proc x11_font_get_xfont*(font: gdk2.PFont): gpointer {.cdecl,
+    dynlib: lib, importc: "gdk_x11_font_get_xfont".}
+  proc FONT_XFONT*(font: gdk2.PFont): gpointer =
+    result = x11_font_get_xfont(font)
+  proc font_lookup_for_display*(display: gdk2.PDisplay, id: PXID): gdk2.PFont =
+    result = cast[gdk2.PFont](xid_table_lookup_for_display(display,id))
+
+when not defined(DISABLE_DEPRECATED):
+  proc x11_font_get_xdisplay*(font: gdk2.PFont): xlib.PDisplay {.cdecl,
+    dynlib: lib, importc: "gdk_x11_font_get_xdisplay".}
+  proc x11_font_get_name*(font: gdk2.PFont): cstring {.cdecl,
+    dynlib: lib, importc: "gdk_x11_font_get_name".}
+  proc FONT_XDISPLAY*(font: gdk2.PFont): xlib.PDisplay =
+    result = x11_font_get_xdisplay(font)
+
+  when not defined(MULTIHEAD_SAFE):
```

Comment:
These ``when``'s come from C but have no meaning for the Nim wrapper. I have no idea what ``MULTIHEAD_SAFE`` means or whether I need it and I'm sure other users of gtk2 don't know either.

Reply:
multihead refers to multiple screens / moitors under linux, but yeah I'll take these `when`'s out, in a minute